typing-break/drw-utils.c

```diff
@@ -168,13 +169,15 @@ set_pixmap_background (GtkWidget *window)
 
 	g_object_unref (tile_pixbuf);
 
-	cr = gdk_cairo_create (gtk_widget_get_window (window));
+	cairo_region = cairo_region_create ();
+	gdc = gdk_window_begin_draw_frame (gtk_widget_get_window (window), cairo_region);
+	cr = gdk_drawing_context_get_cairo_context (gdc);
 	gdk_cairo_set_source_pixbuf (cr, tmp_pixbuf, 0, 0);
 	cairo_paint (cr);
 
 	g_object_unref (tmp_pixbuf);
-
-	cairo_destroy (cr);
+	gdk_window_end_draw_frame (gtk_widget_get_window (window), gdc);
+	cairo_region_destroy (cairo_region);
```

Comment:
this looks fairly suspicious, but I guess it might work on x11.  I can't test, because that doesn't seem to work neither on master or on this branch though, and I get a black break screen (with or without compositing enabled).  If you know a way to make that work, I could try again.

Reply:
In ```X11```, ```gdk_screen_is_composed``` returned ```FALSE```, and ```set_pixmap_background``` was not called, so I am unable to test.I replaced it according to the gtk3 help manual

https://docs.gtk.org/gdk3/func.cairo_create.html

Reply:
It works fine for me on current master using compositor and x11, i started it from status-applet.
The UI is black with white/red text in the middle of the monitor.
Graphic card issue?
I am using AMD integrated gpu.
... i will test current state of the branch.

Reply:
@raveit65 yeah I have the same black background, but some what I read it apparently is supposed to draw a pattern over the desktop, not a black window… anyhow, if black is OK, it worked for me as well.
Using i915 on an IntelHD integrated GPU from a Clarkdale-era CPU.

Reply:
This will probably work on X11 anyway, and IIUC nobody sees a difference with master, so I think it's OK to use that.